File: examples/02/fractions.rb
$stdin.each_line do |line|
    if line.strip == "0"
        break
    end
    num, den = line.split(' ')
    numerator, denominator = num.to_i, den.to_i
    if numerator % denominator == 0
        puts "#{numerator / denominator}"
    else
        smallest = numerator
        if numerator >= denominator
            smallest = denominator
        end

        gcd = 1
        (1..smallest).each do |i|
            if numerator % i == 0 and denominator % i == 0
                gcd = i
            end
        end

        if gcd == 1
            puts "#{numerator}/#{denominator}"
        else
            simple_num = numerator / gcd
            simple_den = denominator / gcd
            puts "#{simple_num}/#{simple_den}"
        end
    end
end

File: examples/02/fractions.c
#include <stdio.h>

int main() {
    int numerator;
    int denominator;
    int gcd;

    while(1) {
        scanf("%d ", &numerator);
        if (numerator == 0) {
            break;
        } else {
            scanf("%d\n", &denominator);
            if (numerator % denominator == 0) {
                printf("%d\n", numerator / denominator);
            } else {

                gcd = find_gcd(numerator, denominator);
                
                if (gcd == 1) {
                    printf("%d/%d\n", numerator, denominator);
                } else {
                    int simple_num = numerator / gcd;
                    int simple_den = denominator / gcd;
                    printf("%d/%d\n", simple_num, simple_den);
                }
            }
        }
    }

    return 0;
}

int find_gcd(int num, int den) {
    int smallest;
    if (num >= den) {
        smallest = den;
    } else {
        smallest = num;
    }

    int i;
    int gcd;
    for (i = 1; i <= smallest; i++) {
        if (num % i == 0 && den % i == 0) {
            gcd = i;
        }
    }
    return gcd;
}
